tko-subs.go

```diff
@@ -170,273 +224,149 @@ func githubcreate(domain string) (bool, error) {
+//to properly figure out if there are any dead DNS records
+func scanDomain(domain string, cmsRecords []*CMS, config Configuration) ([]ScanResult, error) {
+	cname, err := getCnameForDomain(domain)
+	if (err != nil) {
+		return nil, err
+	} else {
+		scanResults := checkCnameAgainstProviders(domain, cname, cmsRecords, config)
+		if (len(scanResults) == 0) {
+			err = errors.New(fmt.Sprintf("Cname [%s] found but could not determine provider", cname))
+		}
+		return scanResults, err
+	}
+}
 
+func getCnameForDomain(domain string) (string, error) {
+	var out, errorOutput bytes.Buffer
 	cmd := exec.Command("dig", "CNAME", domain, "+short")
-	var out bytes.Buffer
 	cmd.Stdout = &out
+	cmd.Stderr = &errorOutput
 	err := cmd.Run()
-	Checkiferr(err)
-
-	//Grabbing the output from the DIG command and storing it in the cname variable
-	cname := out.String()
 
-	var tkr tkosubsResult
-	var isVulnerable bool
-
-	//Defining the transport client that we will need to curl domains
-	tr := &http.Transport{
-		Dial: (&net.Dialer{
-			Timeout: 5 * time.Second,
-		}).Dial,
-		TLSHandshakeTimeout: 5 * time.Second,
-		TLSClientConfig:     &tls.Config{InsecureSkipVerify: true},
+	cname := strings.TrimSpace(out.String())
+	if (err != nil) {
+		return "", err
+	} else if (len(errorOutput.String()) > 0) {
+		return "", errors.New(errorOutput.String())
+	} else if (len(cname) == 0) {
+		return "", errors.New("Cname not found")
 	}
+	return cname, nil
+}
 
-	timeout := time.Duration(5 * time.Second)
-	client := &http.Client{
-		Transport: tr,
-		Timeout:   timeout,
-	}
+//Now, for each entry in the data providers file, we will check to see if the output
+//from the dig command against the current domain matches the CNAME for that data provider
+//if it matches the CNAME, we need to now check if it matches the string for that data provider
+//So, we curl it and see if it matches. At this point, we know its vulnerable
+//Next, if we have the -takeover flag set to 1, we will also try to take over the dangling subdomain
+func checkCnameAgainstProviders(domain string, cname string, cmsRecords []*CMS, config Configuration) ([]ScanResult) {
+	transport := &http.Transport{
+		Dial: (&net.Dialer{ Timeout: 10 * time.Second }).Dial,
+		TLSHandshakeTimeout: 10 * time.Second,
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true}}
 
-	//Now, for each entry in the data providers file, we will check to see if the output
-	//from the dig command against the current domain matches the CNAME for that data provider
-	//if it matches the CNAME, we need to now check if it matches the string for that data provider
-	//So, we curl it and see if it matches. At this point, we know its vulnerable
-	//Next, if we have the -takeover flag set to 1, we will also try to take over the dangling subdomain
+	client := &http.Client{Transport: transport, Timeout: time.Duration(10 * time.Second)}
+	var scanResults []ScanResult
```

Comment:
the variations in uppercase and lowercase for the name itself and the struct is making it slightly confusing to read. Do you mind changing the name to something else like sResults or something like that?

Reply:
I think its best to stick with calling variables what they are, you need to be able to identify their purpose without having to read surrounding code. sResult can be interpreted in many ways, though the "scanResult" can definitely be improved with something more meaningful.

Reply:
Fair enough. I am not really a s/w dev and it sounds like you definitely know what you're talking about so this should be fine